app/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataRecord.java

```diff
@@ -1,185 +1,154 @@
 package org.sagebionetworks.bridge.dynamodb;
 
-import org.sagebionetworks.bridge.json.DateTimeJsonDeserializer;
-import org.sagebionetworks.bridge.json.DateTimeJsonSerializer;
-import org.sagebionetworks.bridge.json.JsonUtils;
+import org.sagebionetworks.bridge.json.DateTimeToStringSerializer;
+import org.sagebionetworks.bridge.json.JodaDateTimeDeserializer;
+import org.sagebionetworks.bridge.json.LocalDateToStringSerializer;
 import org.sagebionetworks.bridge.models.healthdata.HealthDataRecord;
+import org.sagebionetworks.bridge.models.healthdata.HealthDataRecordBuilder;
 
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexHashKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.datatype.joda.deser.LocalDateDeserializer;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
 
-@DynamoDBTable(tableName = "HealthDataRecord2")
+/** DynamoDB implementation of {@link org.sagebionetworks.bridge.models.healthdata.HealthDataRecord}. */
+@DynamoDBTable(tableName = "HealthDataRecord3")
 public class DynamoHealthDataRecord implements HealthDataRecord, DynamoTable {
-
-    private static final String GUID_FIELD = "guid";
-    private static final String START_DATE_FIELD = "startDate";
-    private static final String END_DATE_FIELD = "endDate";
-    private static final String VERSION_FIELD = "version";
-    private static final String DATA_FIELD = "data";
-    
-    private String key;
-    private String guid;
-    private long startDate;
-    private long endDate;
     private JsonNode data;
+    private String healthCode;
+    private String id;
+    private DateTime measuredTime;
```

Comment:
Renamed this to createdOn.

Re: long millis vs DateTime - After discussion with the team, we decided it would be simpler for our object model to use long millis, but the JSON serialization should use ISO timestamps.

Re: Time zone. Standard epoch time is always in UTC, and DateTime is backed by epoch time, so regardless of time zone, it always refers to the same instant in time. (Time zone is, in fact, only used for formatting ISO 8601 data and to provide a reference point for the hour.) Ultimately, the the timestamp comes from the client, so it's up to the client to provide a non-ambiguous timestamp.